### ISSUE TITLE
Breaking change of snapshot_loader

### ISSUE THREAD

**Issue by skaarj1989** [skypjack/entt#1023]
**v3.12.0** changed the behavior of `snapshot_loader` (formerly: `.component<>()`, currently: `.get<>()`)
Now a (default constructible) component is emplaced into a `registry` (which triggers `on_construct`) and then deserialization kicks in. Unfortunately, this makes the `on_construct` callback useless (in my use case).

Example:

```cpp
struct RigidBody {
  struct Settings { /* motion type, friction ... */ };
  Settings settings;
  template <class Archive> void serialize(Archive &archive) {
    archive(settings);
  }
};

void initRigidBody(entt::registry &r, entt::entity e) {
  const auto *collisionShape = getCollisionShape(r, e);
  if (!collisionShape) return;

  auto &world = getPhysicsWorld(r);

  auto &rigidBody = r.get<RigidBody>(e);
  const auto &xf = r.get_or_emplace<Transform>(e);
  const auto *body = world.createBody(xf, rigidBody.settings /* uninitialized/default, not so good */, collisionShape);
  // ...
}

entt::registry r;
r.on_construct<RigidBody>().connect<initRigidBody>();

snapshotLoader.get<RigidBody>(archive);
```

**Comment by skypjack** [comment]
Maybe a dup of #1021 actually. I didn't realize the underlying problem is likely the same.

**Comment by skaarj1989** [comment]
By the way, examples in the [wiki](https://github.com/skypjack/entt/wiki/Crash-Course:-entity-component-system#snapshot-loader) use deprecated API `entities` and `component`.

**Comment by skypjack** [comment]
Good catch. Fixed. 👍

**Comment by skypjack** [comment]
Fixed with `v3.12.1` and on the `wip` branch (soon `master`).
